contracts/Airdrop.sol

```diff
@@ -0,0 +1,96 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.11;
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/math/SafeMath.sol";
+import "./IERC20Metadata.sol";
+contract Airdrop is Ownable {
+    using SafeMath for uint;
+
+    event Setting(address receipt,string token,uint amount);
+    event SettingETH(address receipt,uint amount);
+    event Transfer(address receipt,string token,uint amount);
+    event TransferETH(address recept,uint amount);
+
+    //Owner to token address to amount
+    mapping(address => mapping(address => uint)) private balances;
+
+    mapping(address => uint) private ethBalances;
+
+    // modifier
+    modifier onlyNonzeroAddress(address check) {
+        require(check != address(0), "Invalid zero address");
+        _;
+    }
+
+    // All main function
+    function allowETH(address receipt,uint amount) public onlyOwner onlyNonzeroAddress(receipt){
+        ethBalances[receipt]+=amount;
+        emit SettingETH(receipt , amount);
+    }
+
+    function alowERC20(address receipt,address token,uint amount) public 
+                                                                  onlyOwner onlyNonzeroAddress(receipt)onlyNonzeroAddress(token) 
+    {
+        IERC20Metadata erc20token = IERC20Metadata(token);
+        balances[receipt][token]+=amount;
+
+        emit Setting(receipt,erc20token.name(),amount);
+    }
+
+    function claim(address[] memory tokens,uint[] memory amounts, bool eth,uint ethAmount) public {
+        for(uint i=0;i<tokens.length;i++) {
+            require(_erc20transfer(msg.sender, tokens[i], amounts[i]),"Transfer token fail");
+        }
+        if(eth){
+            require(_ethtransfer(payable(msg.sender), ethAmount),"Transfer eth fail");
+        }
+    }
+
+    // Debug function
+    function balanceOf(address receipt,address token) external view returns(uint) {
+        return balances[receipt][token];
+    }
+
+    function balanceETH(address receipt) external view returns(uint) {
+        return ethBalances[receipt];
+    }
+
+
+    //Internal helper function
+    function _erc20transfer(address receipt, address token,uint amount) internal 
+                                                                        onlyNonzeroAddress(receipt) onlyNonzeroAddress(token) 
+                                                                        returns (bool)
+    {
+        require(balances[msg.sender][token]>=amount,"Insufficient balance");
+        IERC20Metadata erc20token = IERC20Metadata(token);
+        // do nothing with 0 amount
+        if(amount > 0) {
+            balances[msg.sender][token]-=amount;
+            erc20token.transfer(receipt, amount);        
+        }
```

Comment:
Maybe instead of doing nothing if amount = 0, you could just add a require amount > 0, so the function reverts and the user knows that his intended outcome did not happen.